storage/files.js

```diff
@@ -72,326 +68,281 @@ function listFiles (name, callback) {
-  console.log('\tdelete BUCKET_NAME FILE_NAME');
-  console.log('\tgetMetadata BUCKET_NAME FILE_NAME');
-  console.log('\tmakePublic BUCKET_NAME FILE_NAME');
-  console.log('\tmove BUCKET_NAME SRC_FILE_NAME DEST_FILE_NAME');
-  console.log('\tcopy BUCKET_NAME SRC_FILE_NAME DEST_BUCKET_NAME DEST_FILE_NAME');
-  console.log('\nExamples:\n');
-  console.log('\tlist my-bucket');
-  console.log('\tlistByPrefix my-bucket /some-folder');
-  console.log('\tlistByPrefix my-bucket /some-folder -');
-  console.log('\tupload my-bucket ./file.txt');
-  console.log('\tdownload my-bucket file.txt ./file.txt');
-  console.log('\tdelete my-bucket file.txt');
-  console.log('\tgetMetadata my-bucket file.txt');
-  console.log('\tmakePublic my-bucket file.txt');
-  console.log('\tmove my-bucket file.txt file2.txt');
-  console.log('\tcopy my-bucket file.txt my-other-bucket file.txt');
-}
-// [END usage]
+// [END copy_file]
+// [END all]
 
 // The command-line program
-var program = {
+var cli = require('yargs');
+
+var program = module.exports = {
   listFiles: listFiles,
-  listFilesWithPrefix: listFilesWithPrefix,
+  listFilesByPrefix: listFilesByPrefix,
   uploadFile: uploadFile,
   downloadFile: downloadFile,
   deleteFile: deleteFile,
   getMetadata: getMetadata,
   makePublic: makePublic,
   moveFile: moveFile,
   copyFile: copyFile,
-  printUsage: printUsage,
-
-  // Executed when this program is run from the command-line
-  main: function (args, cb) {
-    var command = args.shift();
-    if (command === 'list') {
-      this.listFiles(args[0], cb);
-    } else if (command === 'listByPrefix') {
-      this.listFilesWithPrefix(args[0], args[1], args[2], cb);
-    } else if (command === 'upload') {
-      this.uploadFile(args[0], args[1], cb);
-    } else if (command === 'download') {
-      this.downloadFile(args[0], args[1], args[2], cb);
-    } else if (command === 'delete') {
-      this.deleteFile(args[0], args[1], cb);
-    } else if (command === 'getMetadata') {
-      this.getMetadata(args[0], args[1], cb);
-    } else if (command === 'makePublic') {
-      this.makePublic(args[0], args[1], cb);
-    } else if (command === 'move') {
-      this.moveFile(args[0], args[1], args[2], cb);
-    } else if (command === 'copy') {
-      this.copyFile(args[0], args[1], args[2], args[3], cb);
-    } else {
-      this.printUsage();
-    }
+  main: function (args) {
+    // Run the command-line program
+    cli.help().strict().parse(args).argv;
   }
 };
 
+cli
+  .demand(1)
+  .command('list <bucket> [options]', 'List files in a bucket, optionally filtering by a prefix.', {
+    prefix: {
+      alias: 'p',
+      requiresArg: true,
+      type: 'string',
+      description: 'Filter files by a prefix.'
+    },
+    delimiter: {
+      alias: 'd',
+      requiresArg: true,
+      type: 'string',
+      description: 'Specify a delimiter.'
+    }
+  }, function (options) {
+    if (options.prefix !== undefined) {
```

Comment:
I was thinking `prefix` might be an empty string, but I think you're right. I'll change it.